### DISCUSSION TITLE
How to choose the training data

### DISCUSSION THREAD

**Discussion by cui-hq** [deepmodeling/dpgen#1271] [Q&A] [answered]
In my training prosess with dpgen, in the first iteration, I performed MD simulations with the trained model and choose some clusters from my MD trajectories as new training data, but in the second iteration, the systems quickly become terrible when performing MD simulations. the clusters were not electrical neutrality， so I want to know if the clusters with charges influence the model?

**Comment by njzjz** [comment]
> the clusters were not electrical neutrality

In a potential energy model, the total energy is based on the linear combination of atomic energies, i.e.
```math
E = E' + \sum_\alpha n_\alpha E_\alpha
```
where $n_\alpha$ is the atomic number of element $\alpha$ and $E_\alpha$ is the atomic energy. The atomic energy can be a constant since each atom is assumed to have the same electrons. The number of electrons in your system might be different if the cluster is not electrically neutral. (For example, if it contains $\ce{C}$, $\ce{C^-}$ and $\ce{C^+}$  at the same time.)

**Reply by cui-hq** [reply]
OK, so it's necessary to keep the clusters electrically neutral?

**Reply by njzjz** [reply]
No, but it's necessary to make the net charge of the same element consistent.

**Reply by cui-hq** [reply]
Ok, in my system, there are some CH2=CH-CH2+ molecules, (R)3N molecules and Br- ions, when cutting the clusters from my MD trajectories using dpgen, it's impossible to capture  equal numbers of CH2=CH-CH2+ molecules and Br- ions, for example, there are 12  CH2=CH-CH2+ molecules and 10 Br- ions in one cluster, so when calculating the energy of the cluster using cp2k, the charge was set to +2, it's obviously that my cluster contains C and C+ at the same time, according to your reply, I think it's necessary to add 2 Br- ions in the cluster to keep the cluster electrically neutral. I want to know if it's necessary to keep the cluster electrically neutral by addition of the Br- ions?

**Reply by njzjz** [reply, marked as answer]
This is a cutting-edge topic, and there is no standard answer. You can also try the solution as shown in https://doi.org/10.1038/s41597-022-01882-6, where C and C+ are treated as different elements.

**Reply by cui-hq** [reply]
Thank you for your reply!
